api/routes/bot.py

```diff
@@ -3,20 +3,25 @@
 the Discord bot.
 '''
 
-from fastapi import APIRouter
+from typing import Annotated
+
+from fastapi import APIRouter, Depends
 
 from api import models
+from api.routes.auth import get_current_user
+from DiscordBot.DiscordInterface import DiscordInterface
 
 router = APIRouter()
+discord_interface = DiscordInterface()
 
-@router.post('/create')
-async def create_server():
+@router.post('/create', response_model=models.InviteLink)
+async def create_server(
+    _: Annotated[models.User, Depends(get_current_user)],
+    config: models.ServerConfig
+) -> models.InviteLink:
     '''
     Uses the bot to create a Discord server.
     '''
-
-@router.put('/apply')
-async def apply_config(config: models.ServerConfig):
-    '''
-    Applies a server template configuration to an existing server.
-    '''
+    config_dict = config.model_dump()
+    invite_string = await discord_interface.create_guild(config_dict)
```

Comment:
add error handling here

Reply:
If the bot fails or python throws some exception, then FastAPI will return a 500 error. If the input config is incorrect, then this will be caught by the `ServerConfig` when it's validated.